frontend/src/components/menu/ModalsWrapper.tsx

```diff
@@ -0,0 +1,52 @@
+import ModalDropdown from "./ModalDropdown";
+import ModalEditName from "./ModalEditName";
+import ModalAccountDelete from "./ModalAccountDelete";
+
+import {
+    ModalsWrapper,
+} from "./Menu.styled";
+
+const Modals = (props: any) => {
+    const {
+        currentModal,
+        newUsername,
+        hideModals,
+        updateUsername,
+        handleNicknameChange,
+        toggleAccountEditModal,
+        toggleAccountDeleteModal,
+        deleteAccountConfirm,
+        deleteAccountCancel
+    } = props;
+
+    return (
+        <ModalsWrapper>
+            {currentModal === 1 && (
+                <ModalDropdown
```

Comment:
separation of concerns. how is your solution going to work when we might have e.g. 10, 15+ modals which all handle different logic? some modals might have a scroll widget, others multiple buttons, others custom inputs. how would making a huge one-size-fits-all component benefit the codebase instead of having separate components each dealing with its own tasks?

Reply:
After hours of debate, we have a plan, brilliant plan